src/UKHO.S100PermitService.Common/IO/PermitReaderWriter.cs

```diff
@@ -16,69 +17,108 @@ public class PermitReaderWriter : IPermitReaderWriter
         private const string SecondNamespacePrefix = "ns2";
         private const string XmlDeclaration = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n";
         private const string Namespace = "http://www.iho.int/s100/se/5.0";
-        private const string SchemaFolder = "XmlSchema";
-        private const string PermitSchema = "Permit_Schema.xsd";
+        private const string PermitXmlFileName = "PERMIT.XML";
+        private const string SchemaFile = @"XmlSchema\Permit_Schema.xsd";
+        private readonly string _xsdPath = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!, SchemaFile);
 
-        private readonly string _schemaDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!;
-        private readonly IFileSystem _fileSystem;
+        private readonly ISchemaValidator _schemaValidator;
 
-        public PermitReaderWriter(IFileSystem fileSystem)
+        public PermitReaderWriter(ISchemaValidator schemaValidator)
         {
-            _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+            _schemaValidator = schemaValidator ?? throw new ArgumentNullException(nameof(schemaValidator));
         }
 
-        public string ReadPermit(Permit permit)
+        /// <summary>
+        /// Read Xsd version from schema file
+        /// </summary>
+        /// <returns></returns>
+        public string ReadXsdVersion()
         {
+            XmlSchema? schema;
+            using(var reader = XmlReader.Create(_xsdPath))
+            {
+                schema = XmlSchema.Read(reader, null);
+            }
+
+            return schema?.Version[..5] ?? null;
+        }
+
+        /// <summary>
+        /// Create permit zip
+        /// </summary>
+        /// <param name="permits"></param>
+        /// <returns>ZipStream</returns>
+        public MemoryStream CreatePermits(Dictionary<string, Permit> permits)
```

Comment:
Should take an IReadOnlyDictionary<string, Permit> as a parameter, and return an IStream? Otherwise the method is able to change data that belongs to the caller, which it shouldn't do. The caller shouldn't care what type of stream we create.

Reply:
As discussed, Dictionary to IReadOnlyDictionary and MemoryStream to Stream so the caller shouldn't care what type of stream we create.